R/ChromBackend.R

```diff
@@ -0,0 +1,615 @@
+#' @include hidden_aliases.R
+NULL
+
+#' @title Chromatographic data backends
+#'
+#' @aliases ChromBackend-class
+#'
+#' @description
+#'
+#' `ChromBackend` is a virtual class that defines what different backends need
+#' to provide to be used by the `Chromatograms` package and classes.
+#'
+#' The backend should provide access to the chromatographic data which mainly
+#' consists of (paired) intensity and retention time values. Additional
+#' chromatographic metadata such as MS level and precursor and product m/z
+#' should also be provided.
+#'
+#' The core chromatogram variables are:
+#' - `dataOrigin`: optional `character` with the origin of a chromatogram.
+#' - `dataStorage`: `character` defining where the data is (currently) stored.
+#' - `intensity`: `NumericList` with the intensity values of each chromatogram.
+#' - `msLevel`: `integer` defining the MS level of the data.
+#' - `mzMin`: optional `numeric` with the minimal m/z value in case the data
+#'   was extracted from a `Spectra` object.
+#' - `mzMax`: optional `numeric` with the maximal m/z value.
+#' - `precursorMz`: for SRM data, `numeric` with the target m/z of the
+#'   precursor (parent).
+#' - `precursorMzMin`: for SRM data, optional `numeric` with the lower m/z of
+#'   the precursor's isolation window.
+#' - `precursorMzMax`: for SRM data, optional `numeric` with the upper m/z of
+#'   the precursor's isolation window.
+#' - `productMz` for SRM data, `numeric` with the target m/z of the
+#'   product ion.
+#' - `productMzMin`: for SRM data, optional `numeric` with the lower m/z of
+#'   the product's isolation window.
+#' - `productMzMax`: for SRM data, optional `numeric` with the upper m/z of
+#'   the product's isolation window.
+#' - `rtime`: `NumericList` with the retention times of each chromatogram.
+#'
+#' @name ChromBackend
+#'
+#' @param columns For `chromData` accessor: optional `character` with column
+#'     names (chromatogram variables) that should be included in the
+#'     returned `DataFrame`. By default, all columns are returned.
+#'
+#' @param dataOrigin For `filterDataOrigin`: `character` to define which
+#'     chromatograms to keep.
+#'
+#' @param dataStorage For `filterDataStorage`: `character` to define which
+#'     chromatograms to keep.
+#'
+#' @param drop For `[`: not considered.
```

Comment:
What do you mean here by *not considered*? Should we consider it at all, or should it be *not supported* or even *ignored*?

Reply:
I meant ignored. I'll change that.